### ISSUE TITLE
Can't open task log from Gantt view

### ISSUE THREAD

**Issue by KulykDmytro** [apache/airflow#19613]
### Apache Airflow version

2.2.1

### Operating System

Linux 5.4.149-73.259.amzn2.x86_64

### Versions of Apache Airflow Providers

default for 2.2.1

### Deployment

Other 3rd-party Helm chart

### Deployment details

aws eks using own-developed helm chart

### What happened

When trying to open log from gantt view - receiving an exception
```
  File "/home/airflow/.local/lib/python3.9/site-packages/pendulum/parsing/__init__.py", line 177, in _parse_common
    return date(year, month, day)
ValueError: year 0 is out of range
```

due to incorrect query parameter push: no value for `execution_date` pushed
```
/log?dag_id=report_generator_daily&task_id=check_quints_earnings&execution_date=
```

### What you expected to happen

Logs should be available

### How to reproduce

Open dag's `gantt` chart
click on task ribbon
click on `log`
observe an error

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
You're navigating to logs via this modal, right? If so, does the execution exist in the modal header like the screenshot below?

<img width="588" alt="Screen Shot 2021-11-16 at 11 08 49 AM" src="https://user-images.githubusercontent.com/4600967/142032119-0db919fd-ac31-41ed-96c6-373b37982950.png">

**Comment by jedcunningham** [comment]
I wasn't able to reproduce the original error, but the log link was broken and should be fixed in #20121.

**Comment by jedcunningham** [comment]
Ah! #19258 populated `execution_date` in 2.2.2. Mystery solved.
